src/m365/spo/commands/folder/folder-list.ts

```diff
@@ -73,30 +86,100 @@ class SpoFolderListCommand extends SpoCommand {
     }
 
     try {
-      const resp = await this.getFolderList(args.options.webUrl, args.options.parentFolderUrl, args.options.recursive);
+      const folderProperties = await this.getItemCount(args.options.parentFolderUrl, args);
+
+      // +1 since there is a hidden 'Forms' folder
+      const resp = await this.getFolderList(args.options.parentFolderUrl, args, folderProperties.items + 1, 0);
       logger.log(resp);
     }
     catch (err: any) {
       this.handleRejectedODataJsonPromise(err);
     }
   }
 
-  private async getFolderList(webUrl: string, parentFolderUrl: string, recursive?: boolean, folders: FolderProperties[] = []): Promise<FolderProperties[]> {
-    const serverRelativeUrl: string = urlUtil.getServerRelativePath(webUrl, parentFolderUrl);
+  private async getFolderList(parentFolderUrl: string, args: CommandArgs, items: number, index: number, folders: FolderProperties[] = []): Promise<FolderProperties[]> {
```

Comment:
- Let's rename `index` to `skip`
- Let's give `skip` a default value of `0`
- Let's rename the function to `getFolders`
- Let's remove the `folders: FolderProperties[]` reference array. Instead let's create an array within `getFolders`, add all folders to it and return it.